### ISSUE TITLE
Settings to run 1 bot per IP 24/7

### ISSUE THREAD

**Issue by schrobes** [kolton/d2bot-with-kolbot#629]
Hi,

I've been having issues with getting the right settings to run 1 bot per IP address 24/7. I have been doing some experimenting for a week or so and continue to run into restrictions (5 minute R/D's and wait in line to create game delays). My most recent attempt is running a minimum game time of 3.25 minutes 24/7 but seem to get a restriction after 12-24 hours. Does anyone have any tips or recommendations on settings to adjust to achieve this? If you are able to run a bot 24/7 I would be happy to hear what settings you are working with to acheive this.

**Comment by DarkHorseDre** [comment]
I read there is a limit for 20 per hour and 100 Games Per 12 Hours Per Ip

**Comment by schrobes** [comment]
I have been trying that but still get them, I am wondering how long of a break i need to do on my botting to unflag my ips from bnet

**Comment by yopy12** [comment]
Yes they get lifted and i start back botting again, here is a description of what the issues I've had.

 I first started botting 1-2 weeks ago, during which I was running 4 chars in same game @ ~3 min games. I started getting r/d's on that setup within 24 hours of when i first started and they got more persistent, although they were only 5 minute realm downs, i set my r/d delay to 5 mins and kept botting through it. This went on for a couple days to which it then turned into wait in line to create game delays, which at first were an hour, again I kept botting after those went away to which it turned into a 7 day wait in line to create game delay, yes 7 days. About 5 days into that restriction was the recent blizz update which removed the wait in line delay I had so i started back up with botting. At this time, I purchased a couple proxies and I was running 2 chars per IP in ~6 minute games and again I was having r/d issues. Since the 7 day restriction I saw, I have been trying to break from botting on IPs that get r/d and wait atleast 4 hours before playing on said IP again. The most recent test was setting my minimum game time to 10 minutes, runs were about 6-7 minutes, rest of time sitting in lobby. This test was ran on a fresh proxy (same account/cdkey though) and I still got realm down on only a couple of my chars. 

Any suggestions based on the issues I've had? I am going to try to wait atleast 24 hours before logging back onto bnet on any of the accounts/IPs.

**Comment by schrobes** [comment]
Yes they get lifted and i start back botting again, here is a description of what the issues I've had.

 I first started botting 1-2 weeks ago, during which I was running 4 chars in same game @ ~3 min games. I started getting r/d's on that setup within 24 hours of when i first started and they got more persistent, although they were only 5 minute realm downs, i set my r/d delay to 5 mins and kept botting through it. This went on for a couple days to which it then turned into wait in line to create game delays, which at first were an hour, again I kept botting after those went away to which it turned into a 7 day wait in line to create game delay, yes 7 days. About 5 days into that restriction was the recent blizz update which removed the wait in line delay I had so i started back up with botting. At this time, I purchased a couple proxies and I was running 2 chars per IP in ~6 minute games and again I was having r/d issues. Since the 7 day restriction I saw, I have been trying to break from botting on IPs that get r/d and wait atleast 4 hours before playing on said IP again. The most recent test was setting my minimum game time to 10 minutes, runs were about 6-7 minutes, rest of time sitting in lobby. This test was ran on a fresh proxy (same account/cdkey though) and I still got realm down on only a couple of my chars. 

Any suggestions based on the issues I've had? I am going to try to wait atleast 24 hours before logging back onto bnet on any of the accounts/IPs.

**Comment by DarkHorseDre** [comment]
Nothing I can think of but I've seen the bot do things like retry after R/D too soon (so I extended the retry delay) and other weird stuff.. I've been RD when playing manually due to client bugs - like pressing escape after exiting a game and the game stalls then I get a RD!  I watch mine to see anything weird, chickens, disconnects, whatever. it would be good to see what happens at the time of the RD - is it after a complete set of runs? is it after spamming skills/certain bosses? etc

oh, and are these public games? I assumed not until now.
(I also assume you're using one bnet account per bot)

Have you changed any of the retry timings in lead/follow dbj files?

is Config.PingQuit > 0?

Are the min game times the same for each toon? (If you have one using leader entry script and the rest using follower, this wont matter as the followers join leaders game)

**Comment by schrobes** [comment]
So I have seen the R/D happen before, it is always directly after leaving a game.  These have all been private games too, my chars don't utilize in game chat to communicate. Yes 1 account per bot. I haven't tried changing any of the retry timings, can you indicate what I need to search for in these files to adjust and what exactly they do? I haven't heard of the config.pingquit either, can you explain this too?

Yes i crosschecked the min game times they are all the same, although I do use a leader/follower script, basically I am running a 3 char squad on classic.

**Comment by DarkHorseDre** [comment]
ok good everything seems to be 'normal' lol

config.pingquit is in the character configs - makes the bots exit if the ping is too long - good for hc games - by default is disabled (0)

check `D2BotLead.dbj` and `D2BotFollow.dbj` - they have configurable settings at the top, most of which are timers.  check those and maybe extend them to match the temp ban timings we mentioned.
check them all but mainly the realmdown ofc.

observe the bots to see if there is a pattern - ie if its after a certain amount of time, a certain bot, or script or a certain action (like when they all quit together).  There are delays which can be introduced if needed.

That all said you shouldn't be having these issues so you wouldn't usually need to configure the bot around this - mine didn't have issues when i was using mfhelper.  (are you using mfhelper?)

**Comment by schrobes** [comment]
Reviewed all of those and made some tweaks that should help in certain scenarios (like when i get d2bs is not responding and profile restarts, increased that delay to 7 mins). I am not using mfhelper right now

**Comment by DarkHorseDre** [comment]
cool.
you didn't mention profile restarts before!  I don't get those profile restarts often but any improvement which reduces game exits/starts will help.

I'd definitely watch the bots run, at least until you get an accurate count of games being created.. especially if the profile restarts reset the timers?

fyi I found this, one of the entries about rapid connect/disconnect definitely causes r/d
https://us.battle.net/support/en/article/20457

**Comment by schrobes** [comment]
The profile restarts are not frequent, but it did educate me to edit the crashdelay setting to be higher to prevent lowering the average minutes/game. I am going to wait 48 hours before connecting to bnet again with any of my keys/proxies and go with a 10 minimum game time. Hopefully my scripts aren't causing issues with sending bad packets.

**Comment by DarkHorseDre** [comment]
col.
if you haven't customised them then they wont be sending bad packets. even the packet shopping/casting has been proven to work flawlessly
